### PR TITLE
Migration guide fixes

### PR DESCRIPTION
**Pull request by bugadani**

### CODE REVIEW

**Comment by bugadani** [comment]
@plaes please take a look whether this explains the change well enough.

File: esp-hal/MIGRATING-0.20.md [hunk 1]
```diff
@@ -44,12 +44,32 @@ Instead of manually grabbing peripherals and setting up clocks, you should now c
...
 
 ```rust
-let pin = Input::new(io.gpio0); // pin will have the type `Input<'some>` (or `Input<'some, AnyPin>` if you want to be explicit about it)
-let pin = Input::new_typed(io.gpio0); // pin will have the type `Input<'some, GpioPin<0>>`
+let pin = Input::new_typed(io.gpio0, Pull::Up); // pin will have the type `Input<'some, GpioPin<0>>`
+```
+
+### Creating an `AnyPin`
+
+Instead of `AnyPin::new`, you can now use the `Pin` trait's `degrade` method to obtain an `AnyPin`.
```
Comment (plaes) [inline esp-hal/MIGRATING-0.20.md:64]: Maybe worth mentioning that the new API is similar how embassy-nrf and embassy-stm32 GPIO API is handled.

**Comment by plaes** [comment]
Nice, that's a bit clearer.

**Review by bjoernQ — APPROVED** [review]
Thanks

**Pull request merged**